api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/authenticate/ForgotPassword.java

```diff
@@ -0,0 +1,321 @@
+    }
+
+    /**
+     * Retrieves a WebappDaoFactory instance from the provided HttpServletRequest.
+     *
+     * @param request The HttpServletRequest for accessing the WebappDaoFactory.
+     * @return A WebappDaoFactory instance for database access.
+     */
+    private WebappDaoFactory getWebappDaoFactory(HttpServletRequest request) {
+        return ModelAccess.on(request).getWebappDaoFactory();
+    }
+
+    /**
+     * Builds a reset password link using the provided email and key, and the VitroRequest information.
+     *
+     * @param email The email address of the user.
+     * @param key   The key for resetting the password.
+     * @param vreq  The VitroRequest object containing request information.
+     * @return The reset password link as a string.
+     */
+    private String buildResetPasswordLink(String email, String key, VitroRequest vreq) {
+        try {
+            String relativeUrl = UrlBuilder.getUrl(RESET_PASSWORD_URL, "user", email, "key", key);
+
+            URL context = new URL(vreq.getRequestURL().toString());
+            URL url = new URL(context, relativeUrl);
+            return url.toExternalForm();
+        } catch (MalformedURLException e) {
+            return "error_creating_password_link";
+        }
+    }
+
+    /**
+     * Calculates the expiration date for a password reset link.
+     *
+     * @return The expiration date as a Date object.
+     */
+    private Date calculateExpirationDate() {
+        Calendar c = Calendar.getInstance();
+        c.add(Calendar.DATE, DAYS_TO_USE_PASSWORD_LINK);
+        return c.getTime();
+    }
+
+    /**
+     * Retrieves the URL for the "Forgot Password" page from the VitroRequest context.
+     *
+     * @param request The VitroRequest object containing request information.
+     * @return The URL for the "Forgot Password" page as a string.
+     */
+    private String getForgotPasswordUrl(VitroRequest request) {
+        String contextPath = request.getContextPath();
+        return contextPath + "/forgot-password";
+    }
+
+    /**
+     * Retrieves the URL for the "Contact" page from the VitroRequest context.
+     *
+     * @param request The VitroRequest object containing request information.
+     * @return The URL for the "Contact" page as a string.
+     */
+    private String getContactUrl(VitroRequest request) {
+        String contextPath = request.getContextPath();
+        return contextPath + "/contact";
+    }
+
+    /**
+     * Checks whether the functionality for resetting passwords is enabled based on configuration properties.
+     *
+     * @param vreq The VitroRequest object for retrieving configuration properties.
+     * @return `true` if the functionality is enabled, `false` otherwise.
+     */
+    private boolean isFunctionalityEnabled(VitroRequest vreq) {
+        String enabled = ConfigurationProperties.getBean(vreq).getProperty("authentication.forgotPassword");
+        return enabled.equalsIgnoreCase("enabled");
```

Comment:
Done.

Reply:
I meant that the way it was before it might throw null pointer exception if you don't have authentication.forgotPassword property in config file on comparison  enabled.equalsIgnoreCase("enabled"); 
Now it will throw it one line before.
I suggest to remove  Objects.requireNonNull  and change the comparison to "enabled".equalsIgnoreCase(enabled).